### DISCUSSION TITLE
Use IOptionsMonitor to reload all configuration at runtime?

### DISCUSSION THREAD

**Discussion by kyrvlasiuk** [Burgyn/MMLib.SwaggerForOcelot#207] [Q&A]
What is the user story on reloading configuration at runtime? I see that this library reloads ocelot's FIleConfiguration using IOptionsMonitor , but not anything else,  e.g. SwaggerOptions. How trivial is it to migrate IOption to IOptionsMonitor? So far we haven't considered other options.
Thank you

**Comment by Burgyn** [comment]
I am after appendix surgery. I'll look into it when I get back from the hospital.

**Reply by kyrvlasiuk** [reply]
Get better. There is no rush. When you look into it, please let me know if it is something that aligns with projects goals and if I can contribute.
Regards

**Reply by Burgyn** [reply]
Hi @kyrvlasiuk,

your contribution is welcome. 
There is no special reason to use `IOption` instead of `IOptionsMonitor`. So if you need it you can implement it 😉
